AK/StringUtils.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <AK/Forward.h>
+
+namespace AK {
+
+enum class CaseSensitivity {
+    CaseInsensitive,
+    CaseSensitive,
+};
+
+namespace StringUtils {
+
+    bool matches(const StringView& str, const StringView& mask, CaseSensitivity = CaseSensitivity::CaseInsensitive);
+
+}
+
+}
+
+using AK::CaseSensitivity;
+using AK::StringUtils::matches;
```

Comment:
Let's not bring this into the global namespace.

Reply:
How about 
```C++
using AK::CaseSensitivity;
namespace StringUtils = AK::StringUtils;
```

Reply:
That would work, sure, but does anything outside of `AK` need to use `StringUtils` anyway? It's just meant to share code between `AK::String` and `AK::StringView`, right?

Reply:
So we don't have to type `AK`,  just as other classes in the `AK` namespace.

Reply:
You don't need to type `AK::` if you are within `namespace AK { }` :)

Reply:
`matches()` could be called outside of the `AK` namespace, just as `String::to_int()`.

Reply:
Right, but `String::to_int()` is a member function of `String` whereas `matches()` is a free function with a very generic name. I'm saying let's not put such a generic name in the global namespace. :)

Reply:
It depends on how we intend to use  the `AK::StringUtils` namespace.
Should we use `AK::StringUtils` just for internal use?
If not, the name of `matches` could be changed to something more specific.